lib/JSON/Schema.pm6

```diff
@@ -375,9 +375,13 @@ class JSON::Schema {
     my class RequiredCheck does Check {
         has Str @.prop;
         method check($value --> Nil) {
-            if $value ~~ Associative && not [&&] $value{@!prop}.map(*.defined) {
-                die X::JSON::Schema::Failed.new:
-                    :$!path, :reason("Object does not have required property");
+            if $value ~~ Associative {
+                for @.prop -> $mandatory_prop {
```

Comment:
The declaration `has Str @.prop` is short for `has Str @!prop; method prop() { @!prop }`. A usage of `@.prop` is equivalent to `$(self.prop)` - that is, a (polymorphic) method call. Most of the time inside of a class, we simply want the attribute, so I'd retain the `@!prop` here.

Reply:
The naming convention usually followed in Perl 6 is to use `-` rather than `_` (not by everyone, but it's done in this module, and local consistency is nice).